packages/db/src/repositories/boost.ts

```diff
@@ -644,6 +655,148 @@ class BoostRepository {
+   * 3) Join with agent_boost_totals to get agent information.
+   * 4) Filter for negative delta amounts (debits/spending) for the specified competition.
+   * 5) Order by creation timestamp descending (most recent first).
+   * 6) Apply limit and offset for pagination.
+   *
+   * Read-Only Operation:
+   * - This method only reads data and does not modify any records.
+   * - Safe to call concurrently with other operations.
+   *
+   * Returns:
+   * - Array of boost allocation records with user ID, wallet, agent ID, amount (positive), and timestamp.
+   * - Empty array if no boost allocations found for the competition.
+   * - Amount is returned as positive bigint (negation of deltaAmount).
+   * - Wallet is returned as Uint8Array (binary format from database).
+   *
+   * Notes:
+   * - Only returns spending records (deltaAmount < 0), converted to positive amounts.
+   * - Results ordered by most recent first (createdAt DESC).
+   * - Supports pagination via limit and offset parameters.
+   *
+   * @param args - The query parameters
+   * @param args.competitionId - ID of the competition context
+   * @param args.limit - Maximum number of records to return
+   * @param args.offset - Number of records to skip
+   * @param tx - Optional database transaction to use for the query
+   * @returns Promise resolving to array of boost allocation records
+   */
+  async competitionBoosts(
+    {
+      competitionId,
+      limit,
+      offset,
+    }: { competitionId: string; limit: number; offset: number },
+    tx?: Transaction,
+  ): Promise<Array<ListCompetitionBoost>> {
+    const executor = tx || this.#db;
+    const results = await executor
+      .select({
+        userId: schema.boostBalances.userId,
+        wallet: schema.boostChanges.wallet,
+        agentId: schema.agentBoostTotals.agentId,
+        agentName: agents.name,
+        agentHandle: agents.handle,
+        // Negate deltaAmount to convert spending records (negative) to positive display amounts.
+        // Safe because WHERE clause ensures deltaAmount < 0, guaranteed by decrease() method
+        // which stores spending as -amount (see line 437).
+        amount: sql<bigint>`-${schema.boostChanges.deltaAmount}`.mapWith(
+          BigInt,
+        ),
+        createdAt: schema.boostChanges.createdAt,
+      })
+      .from(schema.boostChanges)
+      .innerJoin(
+        schema.boostBalances,
+        eq(schema.boostChanges.balanceId, schema.boostBalances.id),
+      )
+      .innerJoin(
+        schema.agentBoosts,
+        eq(schema.boostChanges.id, schema.agentBoosts.changeId),
+      )
+      .innerJoin(
+        schema.agentBoostTotals,
+        eq(schema.agentBoosts.agentBoostTotalId, schema.agentBoostTotals.id),
+      )
+      .innerJoin(agents, eq(schema.agentBoostTotals.agentId, agents.id))
+      .where(
+        and(
+          eq(schema.boostBalances.competitionId, competitionId),
+          sql`${schema.boostChanges.deltaAmount} < 0`,
```

Comment:
This is probably fine to leave, but I don't think it's needed. Since you are `innerJoin`ing to `schema.agentBoosts`, that implies that all rows are associated with a boost, and therefore a negative boost change.